test/test_alexa_app_launch_request.js

```diff
@@ -0,0 +1,104 @@
+/*jshint expr: true*/
+"use strict";
+var chai = require("chai");
+var chaiAsPromised = require("chai-as-promised");
+var mockHelper = require("./helpers/mock_helper");
+chai.use(chaiAsPromised);
+var expect = chai.expect;
+chai.config.includeStack = true;
+
+describe("Alexa", function() {
+  var Alexa = require("../index");
+  describe("app", function() {
+    var app = new Alexa.app("myapp");
+    describe("#request", function() {
+      describe("response", function() {
+        var mockRequest = mockHelper.load("intent_request_launch.json");
+
+        context("with an intent request of launchIntent", function() {
+          context("with no launch handler", function() {
+            var app = new Alexa.app("myapp");
+            var subject = app.request(mockRequest);
+            describe("outputSpeech", function() {
+              subject = subject.then(function(response) {
+                return response.response.outputSpeech;
+              });
+              it("responds with NO_LAUNCH_FUNCTION message", function() {
+                return expect(subject).to.eventually.become({
+                  ssml: "<speak>" + app.messages.NO_LAUNCH_FUNCTION + "</speak>",
+                  type: "SSML"
+                });
+              });
+            });
+          });
+
+          context("with a matching intent handler", function() {
+            var app = new Alexa.app("myapp");
+            var expectedMessage = "tubular!";
+            var expectedReprompt = "sweet reprompt dude!";
+            describe("outputSpeech", function() {
+
+              it("handles reprompting correctly", function() {
+                app.launch(function(req, res) {
+                  res.say(expectedMessage).say(expectedMessage).reprompt(expectedReprompt);
+                });
+                var subject = app.request(mockRequest).then(function(response) {
+                  return response.response.outputSpeech;
+                });
+                return expect(subject).to.eventually.become({
+                  ssml: "<speak>" + expectedMessage + " " + expectedMessage + "</speak>",
+                  type: "SSML"
+                });
+              });
+
+              it("combines multiple reprompts?", function() {
+                app.launch(function(req, res) {
+                  res.say(expectedMessage)
+                    .reprompt(expectedReprompt)
+                    .reprompt(expectedReprompt);
+                });
+                var subject = app.request(mockRequest).then(function(response) {
+                  return response.response.reprompt.outputSpeech;
+                });
+                return expect(subject).to.eventually.become({
+                  ssml: "<speak>" + expectedReprompt + "</speak>",
+                  type: "SSML"
+                });
```

Comment:
this result also seemed strange - i saw in the code path logic for combining reprompts but i didn't observe that behavior.